### ISSUE TITLE
progress bar showing confusing status during a failed copy

### ISSUE THREAD

**Issue by DennisDenuto** [carvel-dev/imgpkg#186]
**What steps did you take:**

```
# /home/carvel-imgpkg-v12/cmd/imgpkg/imgpkg copy --tar <redacted>.tar --to-repo <redacted> --registry-username=<redacted> --registry-password=<redacted>
copy | importing 14 images...
 858.78 MiB / 858.80 MiB [===============================================================================================================================================================================================================] 100.00% 27.98 MiB/s
 copy | Error: Error uploading images: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=1575 with Body length 0
 858.78 MiB / 858.80 MiB [===============================================================================================================================================================================================================] 100.00% 11.57 MiB/s
 copy | Error: Error uploading images: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=1575 with Body length 0
 858.78 MiB / 858.80 MiB [================================================================================================================================================================================================================] 100.00% 7.31 MiB/s
 copy | Error: Error uploading images: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=1575 with Body length 0
 858.78 MiB / 858.80 MiB [================================================================================================================================================================================================================] 100.00% 4.89 MiB/s
 copy | Error: Error uploading images: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=2830 with Body length 0
 858.79 MiB / 858.80 MiB [================================================================================================================================================================================================================] 100.00% 3.77 MiB/s
 copy | Error: Error uploading images: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=2830 with Body length 0
 858.79 MiB / 858.80 MiB [==========================================================================================================================================================================================================] 100.00% 3.75 MiB/s 3m49s
copy | done uploading images
Error: Retried 5 times: Put "http://<redacted>/manifests/sha256-<redacted>.imgpkg": http: ContentLength=2830 with Body length 0
# /home/carvel-imgpkg-v12/cmd/imgpkg/imgpkg version
imgpkg version 0.12.0
Succeeded
```

**What happened:**
The progress bar output is showing 100% when the copy command failed due to a timeout.

This *could* be a rounding issue and the copy command was *really really* close to finishing (858.79 MiB / 858.80 MiB) although that could also be reporting incorrectly too.

**What did you expect:**
The progress bar output to show a progress of less than 100% since the copy command did not succeed.

**Anything else you would like to add:**
Looks like the ResponseHeaderTimeout set to 10 seconds in the file registry.go could be the cause, as the client is abruptly closing the connection. Setting to a higher value (30 seconds) or not setting the value, resolves the issue

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by aaronshurley** [comment]
Doing some quick math,
858.78 / 858.80 = 0.9999883558

So, that does round to 100.00 if we're only using two decimals.

I'm thinking that this really isn't a bug unless we don't want to show 100% _only when_ it has been completed. I'm not sure what the technical implications that would have.

@DennisDenuto How would you like to move forward on this? Keep this open, close it out? If we keep it open, I suspect it would be a pretty low priority unless we hear about more folks running into this.

**Comment by DennisDenuto** [comment]
I was more sus about the reported upload in MB (`858.78 / 858.80` and then later `858.79 / 858.80`). was it *really* only .02 MB from successfully writing everything, and managed to make `.01` MB progress before failing again??

I am kinda worried that when a retry occurs, it is messing with the progress bar values (the actual upload isn't really `858.78` but something else). 

moving forward on *this* bug I think would be:

- confirm that retry doesn't mess with the progress bar values.

Note:
A lot of the retry / progress bar logic will be changed as part of https://github.com/vmware-tanzu/carvel-imgpkg/issues/155 (waiting on a new ggcr release to come out)
so this concern will probably be validated/invalidated as part of working on that issue too.

**Comment by DennisDenuto** [comment]
Introduced an error during the 'writing' of an image to force a retry.

The progress bar resets correctly back to zero.

Closing this issue as i'm unable to reproduce this bug.
